dev/sg/internal/images/images.go

```diff
@@ -232,43 +243,90 @@ type imageRepository struct {
 }
 
 func (image ImageReference) String() string {
-	return fmt.Sprintf("%s/%s:%s@%s", image.Registry, image.Name, image.Tag, image.Digest)
+	switch {
+	case image.Tag == "" && image.Digest == "":
+		return fmt.Sprintf("%s/%s", image.Registry, image.Name)
+
+	case image.Tag == "" && image.Digest != "":
+		return fmt.Sprintf("%s/%s@%s", image.Registry, image.Name, image.Digest)
+
+	case image.Tag != "" && image.Digest != "":
```

Comment:
```suggestion
	case image.Tag != "" && image.Digest == "":
```

tiny bug where digest won't be included